test/new-e2e/cws/e2e_cws_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2016-present Datadog, Inc.
+
+package cws
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	cws "github.com/DataDog/datadog-agent/test/new-e2e/cws/lib"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner/parameters"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+)
+
+type agentSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+	apiClient     cws.MyAPIClient
+	signalRuleID  string
+	agentRuleID   string
+	dirname       string
+	filename      string
+	testID        string
+	desc          string
+	agentRuleName string
+	policies      string
+}
+
+func TestAgentSuite(t *testing.T) {
+	agentConfig := "hostname: momar-e2e-test-host\nlog_level: DEBUG\nlogs_enabled: true"
+	e2e.Run(t, &agentSuite{}, e2e.AgentStackDef(
+		[]ec2params.Option{
+			ec2params.WithName("cws-e2e-tests"),
+		},
+		agentparams.WithAgentConfig(agentConfig),
+		agentparams.WithSecurityAgentConfig("runtime_security_config:\n  enabled: true"),
+		agentparams.WithSystemProbeConfig("system_probe_config:\n  enabled: true\n  log_level: trace\nruntime_security_config:\n  enabled: true\nlog_patterns:\n  - module.APIServer.*"),
+		agentparams.WithVersion("7.46.0"),
+	), params.WithDevMode())
+}
+
+func (a *agentSuite) SetupSuite() {
+	// Create temporary directory
+	tempDir := a.Env().VM.Execute("mktemp -d")
+	a.dirname = strings.TrimSuffix(tempDir, "\n")
+	a.filename = fmt.Sprintf("%s/secret", a.dirname)
+	a.testID = uuid.NewString()[:4]
+	a.desc = fmt.Sprintf("e2e test rule %s", a.testID)
+	a.agentRuleName = fmt.Sprintf("e2e_agent_rule_%s", a.testID)
+	a.Suite.SetupSuite()
+}
+
+func (a *agentSuite) TearDownSuite() {
+
+	if len(a.signalRuleID) != 0 {
+		a.apiClient.DeleteSignalRule(a.signalRuleID)
+	}
+	if len(a.agentRuleID) != 0 {
+		a.apiClient.DeleteAgentRule(a.agentRuleID)
+	}
+	a.Env().VM.Execute(fmt.Sprintf("rm -r %s", a.dirname))
+	a.Suite.TearDownSuite()
+}
+
+func (a *agentSuite) TestOpenSignal() {
+	a.apiClient = cws.NewAPIClient()
+
+	// Create CWS Agent rule
+	rule := fmt.Sprintf("open.file.path == \"%s\"", a.filename)
+	res, err := a.apiClient.CreateCWSAgentRule(a.agentRuleName, a.desc, rule)
+	require.NoError(a.T(), err, "Agent rule creation failed")
+	a.agentRuleID = res.Data.GetId()
+
+	// Create Signal Rule (backend)
+	res2, err := a.apiClient.CreateCwsSignalRule(a.desc, "signal rule for e2e testing", a.agentRuleName, []string{})
+	require.NoError(a.T(), err, "Signal rule creation failed")
+	a.signalRuleID = res2.GetId()
+
+	// Check if the agent has started
+	isReady := a.Env().Agent.IsReady()
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
You'd need to retry this, as we don't know when the agent actually start. We have a private method that you can move to public to `WaitForAgentReady`, could be the first thing you do in your test